tests/kip103_test.go

```diff
@@ -0,0 +1,207 @@
+	return apiBackend.Call(ctx, arg, rpc.NewBlockNumberOrHashWithNumber(bn))
+}
+
+func TestRebalanceTreasury_EOA(t *testing.T) {
+	log.EnableLogForTest(log.LvlError, log.LvlInfo)
+
+	fullNode, node, validator, _, workspace := newBlockchain(t, nil)
+	defer func() {
+		fullNode.Stop()
+		os.RemoveAll(workspace)
+	}()
+
+	optsOwner := bind.NewKeyedTransactor(validator.Keys[0])
+	transactor := &testKip103TxTransactor{node: node}
+	targetBlockNum := new(big.Int).Add(node.BlockChain().CurrentBlock().Number(), big.NewInt(10))
+
+	contractAddr, _, contract, err := kip103.DeployTreasuryRebalance(optsOwner, transactor, targetBlockNum)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// set kip103 hardfork config
+	node.BlockChain().Config().KIP103 = &params.KIP103Config{
+		Kip103CompatibleBlock: targetBlockNum,
+		Kip103ContractAddress: contractAddr,
+	}
+
+	t.Log("ContractOwner Addr:", validator.GetAddr().String())
+	t.Log("Contract Addr:", contractAddr.String())
+	t.Log("Target Block:", targetBlockNum.Int64())
+
+	// naive waiting for tx processing
+	time.Sleep(2 * time.Second)
+
+	// prepare newbie accounts
+	numNewbie := 3
+	newbieAccs := make([]TestAccount, numNewbie)
+	newbieAllocs := make([]*big.Int, numNewbie)
+
+	state, err := node.BlockChain().State()
+	if err != nil {
+		t.Fatal(err)
+	}
+	totalNewbieAlloc := state.GetBalance(validator.Addr)
+	t.Log("Total Newbie amount: ", totalNewbieAlloc)
+
+	for i := 0; i < numNewbie; i++ {
+		newbieAccs[i] = genKlaytnLegacyAccount(t)
+		newbieAllocs[i] = new(big.Int).Div(totalNewbieAlloc, big.NewInt(2))
+		totalNewbieAlloc.Sub(totalNewbieAlloc, newbieAllocs[i])
+
+		t.Log("Newbie", i, "Addr:", newbieAccs[i].GetAddr().String())
+		t.Log("Newbie", i, "Amount:", newbieAllocs[i])
+	}
+
+	// register RegisterRetired
+	if _, err := contract.RegisterRetired(optsOwner, validator.Addr); err != nil {
+		t.Fatal(err)
+	}
+
+	// register newbies
+	for i, newbie := range newbieAccs {
+		if _, err := contract.RegisterNewbie(optsOwner, newbie.GetAddr(), newbieAllocs[i]); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// initialized -> registered
+	if _, err := contract.FinalizeRegistration(optsOwner); err != nil {
+		t.Fatal(err)
+	}
+
+	// approve
+	if _, err := contract.Approve(optsOwner, validator.GetAddr()); err != nil {
+		t.Fatal(err)
+	}
+
+	// registered -> approved
+	if _, err := contract.FinalizeApproval(optsOwner); err != nil {
+		t.Fatal(err)
+	}
+
+	// naive waiting for tx processing
+	time.Sleep(2 * time.Second)
+
+	// naive kip103 execution check
+	for i := 0; i < 10; i++ {
+		block := node.BlockChain().CurrentBlock()
+
+		curState, err := node.BlockChain().StateAt(block.Root())
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		balRetired := curState.GetBalance(validator.GetAddr())
+		t.Log("Retired:", balRetired)
+
+		for j, newbie := range newbieAccs {
+			balNewbie := curState.GetBalance(newbie.GetAddr())
+			t.Log("Newbie", j, ":", balNewbie)
+		}
+
+		if block.Number().Cmp(targetBlockNum) == 0 {
```

Comment:
I know it's unlikely but this branch may not be taken at all, in which case it should panic.
Also, I'm not sure why there is a loop.
Consider this suggestion in later PR (we don't have time to fix in this PR):

Suggestion for the entire loop (L177-206):
```
header = waitBlock(node.BlockChain(), targetBlockNum)
if header is nil:
  then panic

assert balance of retired at targetBlockNum == 0
for each newbie {
  assert balance of newbie at targetBlockNum == amount
}
```

Reply:
Apped. 
Actually, I tried to introduce more verification in the test, and wanted to watch past/future values manually. That is why i introduced a loop and remained log statements